### DISCUSSION TITLE
VanX.reactive oldVal?

### DISCUSSION THREAD

**Discussion by HEAVYPOLY** [vanjs-org/van#296] [Q&A] [answered]
Hi, I'm not sure if I'm making a mistake here, how do you access oldVal with vanX.reactive?
const st = vanX.reactive({
page: 'paint',
params: 'abs',
})

  text += `page: ${st.page} | `
  text += `prevpage: ${st.page.oldVal} | `
  
  st.page.oldVal always returns undefined
  
  also tried st.oldVal.page

**Comment by Tao-VanJS** [comment, marked as answer]
You can use `vanX.stateFields` to access the underlying fields.

**Comment by HEAVYPOLY** [comment]
perfect, thank you
